cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -98,7 +94,17 @@ func (pe *ProbeEvent) deleteDiskEvent(msg controller.EventMessage) {
 			mismatch = true
 			continue
 		}
-		pe.Controller.DeactivateDisk(*oldDr)
+		if diskDetails.DiskType == libudevwrapper.UDEV_DISK {
+			pe.Controller.DeactivateDisk(*oldDr)
+		} else if diskDetails.DiskType == libudevwrapper.UDEV_PARTITION {
+			glog.Info("Removing Partition Data from ", diskDetails.ProbeIdentifiers.Uuid)
+			newDrCopy := oldDr.DeepCopy()
+			if len(newDrCopy.Spec.PartitionDetails) > 0 {
+				// Currently the details of last partition will be removed
```

Comment:
How you would like to handle deleting a specific partition, please explain it. Also put TODO in comment section so that it is in our RADAR to fix.

Reply:
For deleting a specific partition we need to have a UID for each partition, so that it can be selected and deleted.

Reply:
We can add partition UID into partition info so that we would be able to identify right partition and delete that partition only. 
Regarding how to generate UID of partition, I am think if we can appen partition number to disk UID. For example :-
disk(sdb) UID = xxxxxxAAAAA
Partition (sdb1) UID = xxxxxxAAAAA-1
Partition (sdb2) UID = xxxxxxAAAAA-2
That way when a delete event comes, we can findout which partition info need to be removed from disk-partition info.